processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -409,6 +409,7 @@ public void close()
   {
     if (!closed) {
       closed = true;
+      threadLocalGrouper.remove();
```

Comment:
Is this going to solve the problem? If I understand correctly, `close()` will get called one time (by the thread that is reading aggregated values out of the ConcurrentGrouper) but none of those threads actually own a ThreadLocal Grouper (the threads that own those Groupers are processing-pool threads).

Reply:
Yes,you are right,it's my fault

Reply:
Can we use  hashmap(ConcurentHashMap<Thread,SpillingGrouper>) instead of threadlocal?
Another way Add a method that is responsible for calling threadLocalGrouper.remove,it can be called in the processing thread

Reply:
Doing our own map, instead of ThreadLocal, sounds like the best solution to me.

Reply:
(Please make sure to include a comment about why we're doing it, since readers might think 'why not use ThreadLocal')